components/ShoppingItem/ShoppingItem.js

```diff
@@ -1,37 +1,53 @@
-import styled from "styled-components";
+import styled, { ThemeConsumer } from "styled-components";
 import Link from "next/link";
 import { useState } from "react";
 
 export default function ShoppingItem({
-  id,
   onDeleteItem,
   shoppingItem,
   onEditItem,
   onChangeMode,
-  mode,
+  onToggleIsPurchased,
+  purchasedItems,
```

Comment:
Well, I do use purchasedItems in the conditionally rendering a label...

And as I have decided for now not to include onChange in the edit function, I do need to pass it down because I use it in the shoppingItemsIndex.

Reply:
I will keep it for now.

Reply:
Please, see my comments below regarding this two issues.

```suggestion

```

Reply:
- I have removed purchasedItems plus the line containing it

- I do need onChange mode here though; esp. as I have now removed onEdit from the ShoppingItem component (see earlier comment)